lib/fs.js

```diff
@@ -1621,8 +1625,15 @@ function ReadStream(path, options) {
     options = {};
   else if (typeof options === 'string')
     options = { encoding: options };
-  else if (options === null || typeof options !== 'object')
-    throw new TypeError('options must be a string or an object');
+  else if (options === null || typeof options !== 'object') {
+    //throw new TypeError('options must be a string or an object');
+    warnedReadStreamOptions = internalUtil.printDeprecationMessage(
+      'Passing anything but an object or a string as the second argument to ' +
+      'ReadStream is deprecated.',
+      warnedReadStreamOptions
+    );
+    options = options || {};
```

Comment:
@chrisdickinson, functions are not supported as option bags in other places (e.g. in [readFileSync](https://github.com/nodejs/io.js/blob/88c1831ff4f7c636832b7c98333e326ed81a7080/lib/fs.js#L389)), so I think they shouldn't be supported here either.